### PR TITLE
Add Food Classes for each Category, also, List all Food by Storage Location

### LINKED ISSUE #83: Convert food category into classes

### LINKED ISSUE #84: As a user, I can list all food stored in a part of the fridge,
...so that I can find food the food I have stored in that location.

### PR DESCRIPTION
**Pull request by Leeyp**
V2.0 tasks

### CODE REVIEW

File: src/main/java/seedu/fridgefriend/command/ListCommand.java [hunk 1]
```diff
@@ -74,27 +106,54 @@ public String getListAllMessage() {
     private String getFoodDescription(int index) {
         int indexShownToUser = index + EXTRA_INDEX;
         Food food = fridge.getFood(index);
-        String foodDescription = 
-                "\n\t" + indexShownToUser +  ". "
-                + food.getFoodName() + " ["
-                + food.getCategory() + "]";
+        String foodDescription =
+                "\n\t" + indexShownToUser + ". "
+                        + food.getFoodName() + " ["
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/command/ListCommand.java]: Should there not be space here?

File: src/main/java/seedu/fridgefriend/command/ListCommand.java [hunk 2]
```diff
@@ -74,27 +106,54 @@ public String getListAllMessage() {
...
+                        + food.getCategory() + "]";
         return foodDescription;
     }
 
-    private void checkIsValidCategory() throws InvalidInputException {
-        if (!FoodCategory.contains(category)) {
-            String errorMessage = "Sorry my friend, please enter a valid food category.";
-            throw new InvalidInputException(errorMessage);
+    private boolean checkIsValidCategory() {
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/command/ListCommand.java:116]: I think you can just return FoodCategory.contains(description) here

File: src/main/java/seedu/fridgefriend/command/ListCommand.java [hunk 3]
```diff
@@ -74,27 +106,54 @@ public String getListAllMessage() {
...
+        if (FoodCategory.contains(description)) {
+            return true;
+        } else {
+            return false;
         }
     }
 
+    private boolean checkIsValidStorageLocation() {
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/command/ListCommand.java:120]: Same here, just return FoodStorageLocation.contains(descrption)

**Review by kwokyto — APPROVED** [review]
Other than the issues in the comments, LGTM

File: src/main/java/seedu/fridgefriend/utilities/Parser.java [hunk 4]
```diff
@@ -33,12 +33,12 @@
      * @param input user's raw input
      * @return Command object
      * @throws EmptyDescriptionException if the required description field is empty
-     * @throws InvalidInputException if the command is not recognised
-     * @throws InvalidIndexException if the index given in description is out of bounds
-     * @throws InvalidDateException if the date input cannot be parsed
+     * @throws InvalidInputException     if the command is not recognised
```
Comment (SimJJ96) [inline src/main/java/seedu/fridgefriend/utilities/Parser.java]: Is extra space allow here?

**Review by SimJJ96 — APPROVED** [review]
LGTM

File: src/main/java/seedu/fridgefriend/command/ListCommand.java (hunk 2 above)
Comment (Leeyp) [reply src/main/java/seedu/fridgefriend/command/ListCommand.java:116]: Hmm I think not necessary? better to abstract/SLAP out leh

File: src/main/java/seedu/fridgefriend/command/ListCommand.java (hunk 3 above)
Comment (Leeyp) [reply src/main/java/seedu/fridgefriend/command/ListCommand.java:124]: same here, I think better to separate out

File: src/main/java/seedu/fridgefriend/command/ListCommand.java (hunk 1 above)
Comment (Leeyp) [reply src/main/java/seedu/fridgefriend/command/ListCommand.java]: Good eye, resolved

File: src/main/java/seedu/fridgefriend/utilities/Parser.java (hunk 4 above)
Comment (Leeyp) [reply src/main/java/seedu/fridgefriend/utilities/Parser.java]: It looks nice on my compiler, can double check?

image

Comment (kwokyto) [reply src/main/java/seedu/fridgefriend/utilities/Parser.java]: i think they trim by default, best to just remove the extra spaces

File: src/main/java/seedu/fridgefriend/command/ListCommand.java [hunk 5]
```diff
@@ -74,27 +106,54 @@ public String getListAllMessage() {
     private String getFoodDescription(int index) {
         int indexShownToUser = index + EXTRA_INDEX;
         Food food = fridge.getFood(index);
-        String foodDescription = 
-                "\n\t" + indexShownToUser +  ". "
-                + food.getFoodName() + " ["
-                + food.getCategory() + "]";
+        String foodDescription =
+                "\n\t" + indexShownToUser + ". "
+                       + food.getFoodName() + " ["
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/command/ListCommand.java]: there are extra spaces here on line 111 and 112

File: src/main/java/seedu/fridgefriend/utilities/Parser.java [hunk 6]
```diff
@@ -62,22 +62,22 @@ public static Command getCommand(String input)
...
      * Returns the appropriate Command object based on command and description.
-     * 
+     *
      * @param parsedInput string array containing command and description
      * @return Command object
      * @throws EmptyDescriptionException if the required description field is empty
-     * @throws InvalidInputException if the command is not recognised
-     * @throws InvalidIndexException if the index given in description is out of bounds
-     * @throws InvalidDateException if the date input cannot be parsed
+     * @throws InvalidInputException     if the command is not recognised
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Parser.java]: extra spaces here on line 75-77

File: src/main/java/seedu/fridgefriend/utilities/Parser.java [hunk 7]
```diff
@@ -131,8 +131,8 @@ public static Command parseCommand(String[] parsedInput)
      * @param foodDescription the string in the required format of food description
      * @return a new AddCommand for Food
      * @throws EmptyDescriptionException if the description is empty
-     * @throws InvalidInputException if the description cannot parse
-     * @throws InvalidDateException if the date input cannot be parsed
+     * @throws InvalidInputException     if the description cannot parse
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Parser.java]: extra spaces here on line 134-135

File: src/main/java/seedu/fridgefriend/utilities/Parser.java [hunk 8]
```diff
@@ -150,22 +150,22 @@ public static Command parseFoodDescription(String foodDescription)
...
     /**
      * Returns an AddCommand object based on description.
-     * 
+     *
      * @param description description for command
      * @return AddCommand object
      * @throws EmptyDescriptionException if the description is empty
-     * @throws InvalidInputException if the description cannot parse
-     * @throws InvalidDateException if the date input cannot be parsed
+     * @throws InvalidInputException     if the description cannot parse
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Parser.java]: extra spaces here on line 157-158

File: src/main/java/seedu/fridgefriend/utilities/Parser.java [hunk 9]
```diff
@@ -176,11 +176,11 @@ public static Command getListCommand(String description) {
 
     /**
      * Returns a RemoveCommand object based on description.
-     * 
+     *
      * @param description description for command
      * @return RemoveCommand object
      * @throws EmptyDescriptionException if the description is empty
-     * @throws InvalidIndexException if the index given in description is out of bounds
+     * @throws InvalidIndexException     if the index given in description is out of bounds
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Parser.java]: extra space here

File: src/main/java/seedu/fridgefriend/utilities/Parser.java [hunk 10]
```diff
@@ -229,24 +229,24 @@ public static Command getByeCommand() {
 
     /**
      * Parses the description into an integer.
-     * 
+     *
      * @param description description for command
      * @return integer index
      * @throws EmptyDescriptionException if the description is empty
-     * @throws InvalidIndexException if the description is not a number
+     * @throws InvalidIndexException     if the description is not a number
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Parser.java]: extra space here

File: src/main/java/seedu/fridgefriend/command/ListCommand.java (hunk 2 above)
Comment (Leeyp) [reply src/main/java/seedu/fridgefriend/command/ListCommand.java:116]: Good catch, resolved

File: src/main/java/seedu/fridgefriend/command/ListCommand.java (hunk 3 above)
Comment (Leeyp) [reply src/main/java/seedu/fridgefriend/command/ListCommand.java:120]: Good catch, resolved

**Review by kwokyto — APPROVED** [review]
i just rmb seeing other spaces as well, it flags up as a solid green block on the github gui so i could see it quite clearly

**Review by Vinci-Hu — APPROVED** [review]
LGTM

**Pull request merged**